### PR TITLE
[SL-ONLY] Bugfix/tracing log format

### PR DESCRIPTION
**Pull request by lpbeliveau-silabs**
Updated the formatting of the timestamps logs and added a check for empty buffer in tracing shell commands

### CODE REVIEW

File: src/platform/silabs/tracing/SilabsTracing.cpp [hunk 1]
```diff
@@ -15,9 +15,11 @@
  *
  ******************************************************************************/
 #include "SilabsTracing.h"
+#include <cstdio> // Include the necessary header for snprintf
 #include <cstring>
 #include <lib/support/CodeUtils.h>
 #include <lib/support/PersistentData.h>
+#include <string> // Include the necessary header for std::string
```
Comment (jmartinez-silabs) [inline src/platform/silabs/tracing/SilabsTracing.cpp:22]: you have `cstring`, `string` and `cstdio`

I would have expected either `stdio.h. with string.h` or `cstdio with cstring` but not three of them no?

**Review by jmartinez-silabs — APPROVED** [review]

File: src/platform/silabs/tracing/SilabsTracing.cpp [hunk 2]
```diff
@@ -122,6 +124,22 @@ const char * OperationTypeToString(OperationType type)
...
 
+std::string FormatTimeStamp(std::chrono::milliseconds time)
+{
+    auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(time).count();
+    auto s  = ms / 1000;
+    ms %= 1000;
+    auto m = s / 60;
+    s %= 60;
+    auto h = m / 60;
+    m %= 60;
```
Comment (mkardous-silabs) [inline src/platform/silabs/tracing/SilabsTracing.cpp:135]: le header time a des constexpr pour faciler les conversions. Ca vaudrait le coup d'au moins utiliser les constantes.

File: src/platform/silabs/tracing/SilabsTracing.cpp [hunk 3]
```diff
@@ -122,6 +124,22 @@ const char * OperationTypeToString(OperationType type)
     }
 }
 
+std::string FormatTimeStamp(std::chrono::milliseconds time)
+{
+    auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(time).count();
```
Comment (mkardous-silabs) [inline src/platform/silabs/tracing/SilabsTracing.cpp:129]: pas sur de comprendre l'utilite du duraction cast. time est deja un `std::chrono::milliseconds`.

**Review by mkardous-silabs — APPROVED** [review]

File: src/platform/silabs/tracing/SilabsTracing.cpp (hunk 1 above)
Comment (lpbeliveau-silabs) [reply src/platform/silabs/tracing/SilabsTracing.cpp:22]: cstring and cstdio are compatibility includes for C standard library to be included in C++, <string.h> is for C++ string. Let me check if there would be a clean way to ditch C++ string type here...

**Pull request merged**